google/cloud/aiplatform/models.py

```diff
@@ -215,10 +228,483 @@ def upload(
+                decreases, some of these extra replicas may be freed.
+            max_replica_count (int):
+                Optional. The maximum number of replicas this deployed model may
+                be deployed on when the traffic against it increases. If requested
+                value is too large, the deployment will error, but if deployment
+                succeeds then the ability to scale the model to that many replicas
+                is guaranteed (barring service outages). If traffic against the
+                deployed model increases beyond what its replicas at maximum may
+                handle, a portion of the traffic will be dropped. If this value
+                is not provided, the smaller value of min_replica_count or 1 will
+                be used.
+            metadata (Sequence[Tuple[str, str]]):
+                Optional. Strings which should be sent along with the request as
+                metadata.
+        Returns:
+            lro (lro.LRO):
+                LRO of model deployment.
+        """
+        max_replica_count = min(1, min_replica_count)
+        if machine_type:
+            machine_spec = machine_resources.MachineSpec(machine_type=machine_type)
+            dedicated_resources = machine_resources.DedicatedResources(
+                machine_spec=machine_spec,
+                min_replica_count=min_replica_count,
+                max_replica_count=max_replica_count,
+            )
+            deployed_model = gca_endpoint.DeployedModel(
+                dedicated_resources=dedicated_resources,
+                model=model.resource_name,
+                display_name=deployed_model_display_name,
+            )
+        else:
+            automatic_resources = machine_resources.AutomaticResources(
+                min_replica_count=min_replica_count,
+                max_replica_count=max_replica_count,
+            )
+            deployed_model = gca_endpoint.DeployedModel(
+                automatic_resources=automatic_resources,
+                model=model.resource_name,
+                display_name=deployed_model_display_name,
+            )
+
+        if traffic_split is None:
+            if bool(self._gca_resource.traffic_split) is False:
+                raise ValueError(
+                    """There are currently no deployed models so the traffic percentage
+                for this deployed model needs to be 100."""
+                )
+            traffic_split = self._allocate_traffic(
+                traffic_split=dict(self._gca_resource.traffic_split),
+                traffic_percentage=traffic_percentage,
+            )
+        # else:
+        #     TODO: check traffic split provided is valid
+
+        operation_future = self.endpoint_client.deploy_model(
+            endpoint=self.resource_name,
+            deployed_model=deployed_model,
+            traffic_split=traffic_split,
+            metadata=metadata,
+        )
+
+        return lro.LRO(operation_future)
```

Comment:
Block before returning. I also think we don't need to pass back the LRO. This can just return None.

Reply:
why block? will we change this in the future?

Reply:
We will change it in the future. We should keep the experience consistent across resource nouns for now and add async consistently across the SDK.

Reply:
will block.